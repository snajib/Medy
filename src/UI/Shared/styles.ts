import { StyleSheet, ViewStyle, Dimensions, TextStyle } from 'react-native';

export const width = Dimensions.get('window').width;
export const height = Dimensions.get('window').height;

const topPrimaryContainterStyle: ViewStyle = {
  flex: 1,
  backgroundColor: '#5F44BA',
  alignItems: 'center',
};

const topSecondaryContainerStyle: ViewStyle = {
  flex: 1,
  marginTop: '22%',
};

const centerText: ViewStyle = {
  flex: 1,
  alignItems: 'center',
  justifyContent: 'center',
};

const titleText: TextStyle = {
  fontSize: 36,
  fontStyle: 'normal',
  fontWeight: 'bold',
  textAlign: 'left',
  color: '#fff',
  paddingLeft: '3.5%',
};

const wholeDefaultCardStyle: ViewStyle = {
  backgroundColor: '#fff',
  borderRadius: 12,
  marginTop: 10,
  marginLeft: 0,
  marginRight: 0,
  marginBottom: 0,
  height: height,
  width: width,
};

const finderDefaultCardStyle: ViewStyle = {
  backgroundColor: '#fff',
  borderRadius: 12,
  shadowColor: '#000',
  shadowOffset: { width: 0, height: 3 },
  shadowOpacity: 0.8,
  shadowRadius: 3,
  marginTop: 100,
  marginLeft: 10,
  marginRight: 10,
  height: height - 300,
  width: width - 100,
};

const homeSandwichCardStyle: ViewStyle = {
  backgroundColor: '#fff',
  borderRadius: 12,
  shadowColor: '#000',
  shadowOffset: { width: 0, height: 3 },
  shadowOpacity: 0.8,
  shadowRadius: 3,
  height: height - 450,
  width: width - 100,
  marginBottom: 40,
};

const finderSandwichCardStyle: ViewStyle = {
  backgroundColor: '#fff',
  borderRadius: 12,
  shadowColor: '#000',
  shadowOffset: { width: 0, height: 3 },
  shadowOpacity: 0.8,
  shadowRadius: 3,
  height: height - 650,
  width: width - 100,
  marginBottom: 40,
};

const topAccordionContainer: ViewStyle = {
  paddingLeft: '5%',
  paddingTop: '2.5%',
};

const accordionSectionLabel: TextStyle = {
  fontSize: 30,
  fontStyle: 'normal',
  fontWeight: 'bold',
  textAlign: 'left',
  color: '#000',
  paddingTop: '5%',
};

const sectionActive: ViewStyle = {
  paddingLeft: '5%',
};

const sectionInActive: ViewStyle = {
  height: 0,
  overflow: 'hidden',
};

export const styles = StyleSheet.create({
  topPrimaryContainterStyle,
  topSecondaryContainerStyle,
  finderDefaultCardStyle,
  centerText,
  titleText,
  homeSandwichCardStyle,
  finderSandwichCardStyle,
  wholeDefaultCardStyle,
  topAccordionContainer,
  accordionSectionLabel,
  sectionActive,
  sectionInActive,
});
