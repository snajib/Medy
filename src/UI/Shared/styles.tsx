import { StyleSheet, ViewStyle } from "react-native";

const topContainterStyle: ViewStyle = {
   flex: 1,
   backgroundColor: "#5F44BA",
   alignItems: "center",
   justifyContent: "center"
};

const centerText: ViewStyle = {
   flex: 1,
   alignItems: "center",
   justifyContent: "center"
};

const mainCardStyle: ViewStyle = {
   backgroundColor: "#fff",
   borderRadius: 12,
   shadowColor: "#000",
   shadowOffset: { width: 0, height: 3 },
   shadowOpacity: 0.8,
   shadowRadius: 3,
   height: "70%",
   width: "67%"
};

export const styles = StyleSheet.create({
   topContainterStyle,
   mainCardStyle,
   centerText
});
