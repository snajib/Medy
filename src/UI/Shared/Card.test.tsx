import React from 'react';
import renderer from 'react-test-renderer';
import { styles } from './styles';
import { Card, ICardProps } from './card';
import { Text, View } from 'react-native';

const cardProps: ICardProps = {
  profileName: 'Test',
  summary: 'abc',
  key: 'key',
};

it('should render the correct card based on props', () => {
  const card = renderer.create(
    <Card {...cardProps} style={styles.finderDefaultCardStyle} />,
  );
  expect(card.root.props.style.height).toBe(1034);

  const card2 = renderer.create(
    <Card {...cardProps} style={styles.finderSandwichCardStyle} />,
  );
  expect(card2.root.props.style.height).toBe(684);
});

it('should render the correct number of components', () => {
  const card = renderer.create(
    <Card {...cardProps} style={styles.finderDefaultCardStyle} />,
  );
  expect(card.root.findAllByType(Text).length).toBe(2);
  expect(card.root.findAllByType(Text)[0].props.children).toEqual(
    cardProps.profileName,
  );
  expect(card.root.findAllByType(Text)[1].props.children).toEqual(
    cardProps.summary,
  );
  expect(card.root.findAllByType(View).length).toBe(2);
});

it('should render with snapshot', () => {
  const card = renderer.create(
    <Card {...cardProps} style={styles.finderDefaultCardStyle} />,
  );

  const card2 = renderer.create(
    <Card {...cardProps} style={styles.finderSandwichCardStyle} />,
  );

  expect(card.toJSON()).toMatchSnapshot();
  expect(card2.toJSON()).toMatchSnapshot();
});
