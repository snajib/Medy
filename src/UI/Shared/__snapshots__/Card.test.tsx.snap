// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderRadius": 12,
      "height": 1034,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 100,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.8,
      "shadowRadius": 3,
      "width": 650,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text>
      Test
    </Text>
    <Text>
      abc
    </Text>
  </View>
</View>
`;

exports[`should render with snapshot 2`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderRadius": 12,
      "height": 684,
      "marginBottom": 40,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.8,
      "shadowRadius": 3,
      "width": 650,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text>
      Test
    </Text>
    <Text>
      abc
    </Text>
  </View>
</View>
`;
